localstack-core/localstack/services/route53resolver/provider.py

```diff
@@ -371,19 +372,31 @@ def list_firewall_rules(
         next_token: NextToken = None,
         **kwargs,
     ) -> ListFirewallRulesResponse:
-        """List all the firewall rules in a firewall rule group."""
-        # TODO: implement priority and action filtering
+        """List firewall rules in a firewall rule group.
+
+        Rules will be filtered by priority and action if values for these params are provided.
+
+        Raises:
+            ResourceNotFound: If a firewall group by the provided id does not exist.
+        """
+        # TODO: implement max_results filtering and next_token handling
         store = self.get_store(context.account_id, context.region)
-        firewall_rules = []
-        for firewall_rule in store.firewall_rules.get(firewall_rule_group_id, {}).values():
-            firewall_rules.append(FirewallRule(firewall_rule))
-        if len(firewall_rules) == 0:
+        firewall_rule_group = store.firewall_rules.get(firewall_rule_group_id, None)
+        if firewall_rule_group is None:
             raise ResourceNotFoundException(
                 f"Can't find the resource with ID '{firewall_rule_group_id}'. Trace Id: '{localstack.services.route53resolver.utils.get_trace_id()}'"
             )
-        return ListFirewallRulesResponse(
-            FirewallRules=firewall_rules,
-        )
+
+        firewall_rules = []
+        for firewall_rule in firewall_rule_group.values():
+            if action is not None and action != firewall_rule["Action"]:
+                continue
+            if priority is not None and priority != firewall_rule["Priority"]:
+                continue
+
+            firewall_rules.append(FirewallRule(firewall_rule))
```

Comment:
we could improve the structure a bit here with something like the following: 

```
firewall_rules = [
    FirewallRule(rule) 
    for rule in firewall_rule_group.values()
    if (action is None or action == firewall_rule["Action"]) 
    and (priority is None or priority == firewall_rule["Priority"])
]
```

Reply:
I applied @sannya-singal suggestion.